mk/libraries.mk

```diff
@@ -91,7 +91,7 @@ define build-library
     $(1)_PATH := $$(_d)/$$($(1)_NAME).$(SO_EXT)
 
     $$($(1)_PATH): $$($(1)_OBJS) $$(_libs) | $$(_d)/
-	$$(trace-ld) $(CXX) -o $$(abspath $$@) -shared $$(LDFLAGS) $$(GLOBAL_LDFLAGS) $$($(1)_OBJS) $$($(1)_LDFLAGS) $$($(1)_LDFLAGS_PROPAGATED) $$(foreach lib, $$($(1)_LIBS), $$($$(lib)_LDFLAGS_USE)) $$($(1)_LDFLAGS_UNINSTALLED)
+	+$$(trace-ld) $(CXX) -o $$(abspath $$@) -shared $$(LDFLAGS) $$(GLOBAL_LDFLAGS) $$($(1)_OBJS) $$($(1)_LDFLAGS) $$($(1)_LDFLAGS_PROPAGATED) $$(foreach lib, $$($(1)_LIBS), $$($$(lib)_LDFLAGS_USE)) $$($(1)_LDFLAGS_UNINSTALLED)
```

Comment:
Any reason why the + was added ? it make `make -n` ineffective. I don't see it linked with LTO

Reply:
Could it be that the jobserver is not working inside of nix build? I have seen a lot of messages in the logs around that lately.

Reply:
That patch does not apply to gcc12 and for gcc11 the jobserver is still written in c.


I switched to gcc12 and still get a lot of:


```
nix> lto-wrapper: warning: jobserver is not available: '--jobserver-auth=' is not present in 'MAKEFLAGS'
nix> lto-wrapper: note: see the '-flto' option documentation for more information
nix> lto-wrapper: warning: jobserver is not available: '--jobserver-auth=' is not present in 'MAKEFLAGS'
nix> lto-wrapper: note: see the '-flto' option documentation for more information
```

Reply:
If it is as easy as that, we should try that.

Reply:
Following overlay worked for applying the patch:

```nix
  nixVersions = prev.nixVersions // {
    stable = prev.nixVersions.stable.override {
      # stdenv = gcc12Stdenv;
      stdenv = overrideCC gcc12Stdenv (gcc.override {
        cc = gcc12.cc.overrideAttrs (oldAttrs: {
          patches = oldAttrs.patches or [ ] ++ [
            # dependency of patch below
            (fetchpatch {
              url = "https://github.com/gcc-mirror/gcc/commit/1270ccda70ca09f7d4fe76b5156dca8992bd77a6.patch";
              sha256 = "sha256-XkzktcHOuYNejFGjQYkmM+QU7/j3EHM8mUOLuczRSRY=";
            })
            # https://github.com/NixOS/nix/pull/5813#discussion_r1101203213
            # https://gcc.gnu.org/pipermail/gcc-patches/2022-August/599313.html
            (fetchpatch {
              url = "https://github.com/gcc-mirror/gcc/commit/53e3b2bf16a486c15c20991c6095f7be09012b55.patch";
              sha256 = "sha256-5AfTrsayft/bzSNsaTjqR7hB7tDq3UI4PufxQAjKG6c=";
            })
          ];
        });
      });
    }
  };
```